src/antlr/SqlLineageParseTree.ts

```diff
@@ -0,0 +1,145 @@
+import _ from "lodash";
+
+/**
+ * Options available for SQL parser.
+ */
+export interface LineageParserOptions {
+  /**
+   * Whether double quoted identifiers are allowed. If `true` - then both double quotes and backticks can be used
+   * to quote identifiers. String literals are quoted with single quotes only.
+   * If `false` (default) - double quotes are used for string literals (as an alternative
+   * to single quotes). Identifiers are quoted with backquotes.
+   */
+  doubleQuotedIdentifier?: boolean;
+}
+
+/**
+ * SQL parse tree with available operations.
+ */
+export class SqlLineageParseTree {
+  /**
+   * Creates SQL parse tree from antlr StatementContext
+   * @param tree StatementContext object which is the product of parsing SQL
+   * @param cursor A representation of the cursor to look for in the query
+   */
+  constructor(public readonly tree: StatementContext, readonly cursor?: Cursor) {}
+
+  /**
+   * Extracts and returns all potentially used tables. Note that this method does not perform context
+   * analysis and thus can return not only external tables used but also references to CTEs or subqueries
+   * defined inside the query itself. But it is guaranteed that all external (to the query)
+   * tables will be returned.
+   * This method commonly used to analyse query and pre-fetch metadata for tables used.
+   * @returns Tables used in query
+   */
+  getUsedTables(): { references: TablePrimary[]; incomplete: TablePrimaryIncomplete[] } {
+    const visitor = new ExtractTablesVisitor(this.cursor);
+    return this.tree.accept(visitor);
+  }
+
+  /**
+   * Extracts column level lineage from SQL parse tree.
+   * There are 2 principal modes that control lineage representation: "merged leaves" and "tree" (default).
+   * - In "tree" mode (default) all source tables are displayed with all their columns and mentioned as many
+   *   times as they occur in the query.
+   * - In "mergedLeaves" mode source tables are mentioned only once even if they are used multiple times in
+   *   the query. Source table columns that are not used in the query omitted from lineage.
+   * @param getTable Function to get table metadata. It takes table identifier and returns some table data
+   *    plus the list of columns for this table. Columns are expected to be in particular order as defined
+   *    in this table's DDL.
+   * @param mergedLeaves Selects mode for the lineage generation ("tree" (default) when `false`,
+   * "mergedLeaves" when `true`).
+   * @param options Lineage generation options:
+   * - `positionalRefsEnabled` (`false` by default) options controls whether to interpret numerical references
+   * inside ORDER BY as references to SELECT list expressions
+   * @returns Calculated lineage.
+   */
+  getLineage<TableData, ColumnData>(
+    getTable: (
+      id: TablePrimary
+    ) => { table: { id: string; data: TableData }; columns: { id: string; data: ColumnData }[] } | undefined = () =>
+      undefined,
```

Comment:
@megaflo @pavelgubin Added a default `getTable` implementation of
```typescript
() => undefined
```
so that it is more comfortable to use rhombic without any metadata (only relying on the assumed tables and columns).